flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/DynamicKafkaDeserializationSchemaTest.java

```diff
@@ -0,0 +1,199 @@
+                new ConsumerRecord<>(
+                        "",
+                        0,
+                        1L,
+                        1L,
+                        TimestampType.NO_TIMESTAMP_TYPE,
+                        testSpec.valueOnly ? 0 : 1,
+                        1,
+                        key,
+                        value,
+                        new RecordHeaders(),
+                        Optional.empty());
+        try {
+            dynamicKafkaRecordDeserializationSchema.deserialize(message, null);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    static Collection<TestSpec> configProvider() {
+        return ImmutableList.<TestSpec>builder().addAll(getValidTestSpecs()).build();
+    }
+
+    @Test
+    public void testdeserializeWithoutMethod() {
+
+        // ensure we have the method.
+        DynamicKafkaRecordSerializationSchema.nullifySerializeWithAdditionalPropertiesMethod();
+        try {
+            FlinkKafkaPartitioner<RowData> partitioner = null;
+            MockDeserializationSchema<RowData> keyDeserialization =
+                    new MockDeserializationSchema<>(true, false);
+            MockDeserializationSchema<RowData> valueDeserialization =
+                    new MockDeserializationSchema<>(false, false);
+
+            RowData.FieldGetter[] keyFieldGetters = null;
+
+            int[] keyProjection = {0, 1};
+            int[] valueProjection = {2, 3};
+            TypeInformation producedTypeInfo = null;
+            DynamicKafkaDeserializationSchema dynamicKafkaRecordDeserializationSchema =
+                    new DynamicKafkaDeserializationSchema(
+                            4,
+                            keyDeserialization,
+                            keyProjection,
+                            valueDeserialization,
+                            valueProjection,
+                            false,
+                            null,
+                            producedTypeInfo,
+                            false);
+            byte[] key = {0};
+            byte[] value = {0};
+            ConsumerRecord<byte[], byte[]> message =
+                    new ConsumerRecord<>(
+                            "",
+                            0,
+                            1L,
+                            1L,
+                            TimestampType.NO_TIMESTAMP_TYPE,
+                            1,
+                            1,
+                            key,
+                            value,
+                            new RecordHeaders(),
+                            Optional.empty());
+            try {
+                dynamicKafkaRecordDeserializationSchema.deserialize(message, null);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        } finally {
+            DynamicKafkaRecordSerializationSchema.initializeMethod();
```

Comment:
Why are you using `finally` here?

This looks like a side effect on a static object for the test - would it not be better under a `beforeEach()`?